src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -410,17 +413,9 @@ public void before() throws Throwable {
             f.set(null,null);
             throw e;
         }
-        jenkins.setNoUsageStatistics(true); // collecting usage stats from tests are pointless.
-
-        jenkins.setCrumbIssuer(new TestCrumbIssuer());
-
-        jenkins.servletContext.setAttribute("app",jenkins);
-        jenkins.servletContext.setAttribute("version","?");
-        WebAppMain.installExpressionFactory(new ServletContextEvent(jenkins.servletContext));
-
-        // set a default JDK to be the one that the harness is using.
-        jenkins.getJDKs().add(new JDK("default",System.getProperty("java.home")));
 
+        jenkins.setCrumbIssuer(new TestCrumbIssuer());  // TODO: Move to _configureJenkinsForTest after JENKINS-55240
```

Comment:
I asked for information in JIRA and got none. What is the issue?

Reply:
When we use configuration as code to setup the Jenkins instance with `TestCrumbIssuer`, it was refused to be marshaled. This was happening when the code was in the Role Strategy Plugin.

Reply:
> use configuration as code to setup the Jenkins instance

So, `TestCrumbIssuer` not being loaded from the usual location I suppose? Normally it works because of [this clause](https://github.com/jenkinsci/jenkins/blob/c489b07f311eb1c3068d0457ff7813261ec8ece5/core/src/main/java/jenkins/security/ClassFilterImpl.java#L249-L252).